kura/org.eclipse.kura.rest.network.status.provider/src/main/java/org/eclipse/kura/network/status/provider/api/InterfaceStatusListDTO.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.network.status.provider.api;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.kura.net.status.NetworkInterfaceStatus;
+
+@SuppressWarnings("unused")
+public class InterfaceStatusListDTO {
+
+    private final List<NetworkInterfaceStatusDTO> interfaces;
+    private final List<FailureDTO> failures;
+
+    public InterfaceStatusListDTO(final List<NetworkInterfaceStatus> interfaces, final List<FailureDTO> failures) {
+        this.interfaces = interfaces.stream().map(NetworkInterfaceStatusDTO::fromNetworkInterfaceStatus)
+                .collect(Collectors.toList());
+        this.failures = (failures == null || failures.isEmpty()) ? null : failures;
+    }
+
```

Comment:
Setting the filed as null will cause GSON to not serialize it, so the failures property will not be added at all if there are no failures, it will not appear as `"failures": null`. But I don't have a strong preference, if you want we can also always serialize the field as an empty JSON array.

Reply:
I've neither a strong preference. I only prefer to implement the same behavior of other Kura RestAPis. What is the default behavior with the others?

Reply:
We don't have a consistent behavior. Since all other arrays in this resta apis are always serialized we could always serialize also this one even if there are no failures.